### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by jcantosz**
Docker image builds from GH repo. Runs gradlew. Installs node + npm modules for UI

### CODE REVIEW

**Comment by jcantosz** [comment]
Caught this in the compose logs, looks like there is something wrong with my elastic configuration:
`conductor_1      | 8560 [main] ERROR com.netflix.conductor.server.ConductorServer  - Error starting embedded elasticsearch.  Search functionality will be impacted: file:/conductor-server-1.6.0-SNAPSHOT-all.jar!/es_template.json`. 
If anything jumps out at you as to what I configured wrong, please let me know. Does this impact the use of dynomite? I will be taking a closer look at it tomorrow

File: docker/tmp/config.properties [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+# Transport address to elasticsearch
+workflow.elasticsearch.url=localhost:9300
+
+# Name of the elasticsearch cluster
+workflow.elasticsearch.index.name=conductor
+
+# Additional modules (optional)
+conductor.additional.modules=class_extending_com.google.inject.AbstractModule
```
Comment (v1r3n) [inline docker/tmp/config.properties]: You might want to comment this line.

Comment (jcantosz) [inline docker/tmp/config.properties]: Thanks! That appears to have fixed the elasticsearch problem

**Comment by jcantosz** [comment]
Now that elasticsearch is not falling back to the in-memory instance, I am seeing an issue connecting to dynomite. `startup.sh` waits for the dynomite machine to be up and the port 22122 to be open on it before starting the server. I assume I have some configuration wrong here as well. 

Docker output:
```
conductor_1      | com.netflix.dyno.connectionpool.exception.DynoConnectException: DynoConnectException: [host=Host [hostname=UNKNOWN, ipAddress=UNKNOWN, port=0, rack: UNKNOWN, datacenter: UNKNOW, status: Down], latency=0(0), attempts=0]Unable to make ANY successful connections to host Host [hostname=dynomite, ipAddress=null, port=8102, rack: a, datacenter: , status: Up]
```
API Response:
 ```
{
  "code": "INTERNAL_ERROR",
  "message": "INTERNAL_ERROR - PoolOfflineException: [host=Host [hostname=UNKNOWN, ipAddress=UNKNOWN, port=0, rack: UNKNOWN, datacenter: UNKNOW, status: Down], latency=0(0), attempts=0]host pool is offline and no Racks available for fallback",
  "instance": "f83f92150c88"
}
```

**Comment by v1r3n** [comment]
I have the docker compose working with bit of changes in Conductor's compose configuration and Docker image for dynomite.    Let me merge this PR and apply changes on top of it to get it working.

**Comment by jcantosz** [comment]
@v1r3n Thanks, let me know if there is anything further you need from me WRT these changes. I may open and begin working on a few docker related enhancements I came across when writing this (using published jar, separating conductor server and ui into 2 containers, separating layers for better local caching/building)

**Comment by v1r3n** [comment]
@jcantosz Thanks for the PR, I am going to setup a pipeline that will do automatic publishing to Docker hub for each release.  

Separate docker file for UI a good idea, UI connects to the workflow server running on env variable ```WF_SERVER```.  Other than that  no other configs are required.

One more thing, if you are interested is to make the server module expose more jetty fine grained control for a production ready deployment.  I have some ideas, maybe we can start a project  and discuss on that.  But in the meantime, separate containers is a good one.

**Comment by jcantosz** [comment]
@v1r3n Great, I'll get started on UI/backend split. Thanks for letting me know about the env config.

Yeah, I'd like to hear what you are thinking for jetty configuration, lets take a look at that after this is completed.

**Pull request merged**